### DISCUSSION TITLE
Some questions / suggestions

### DISCUSSION THREAD

**Discussion by xfzv** [fosskers/aura#792] [Ideas]
Hi

I've been playing around with aura lately. I have some questions / suggestions.

1. I found by accident that it's possible to use `XDG_CONFIG_HOME/aura/aura.conf` instead of `/etc/aura.conf`. The [user guide](https://fosskers.github.io/aura/configuration.html?highlight=aura.conf#configuring-aura) and the README only mention `/etc/aura.conf`:

https://github.com/fosskers/aura/blob/b43a71c172d56ad1e7fa3a6090b8711e5199b441/README.md?plain=1#L233-L236

Plus, the link to the template config file is broken.

2. The downgrade feature (`-C`) is very handy. However, it only works with local Pacman cache in `/var/cache/pacman/pkg`. It would be nice if we could also downgrade from [archive.archlinux.org](https://archive.archlinux.org/) like with [downgrade](https://github.com/archlinux-downgrade/downgrade/blob/1f73525b686f0e351b734ac17467e0b564a5b7f0/README.md?plain=1#L46-L48). Useful if the Pacman cache is empty or if we want to downgrade to a much older version for some reasons.

3. No packages completions for `aura -A`? I'm aware of https://github.com/fosskers/aura/issues/106 but I'm not sure what is the current status of this. Works fine with `paru -S` for example.

4. Is there a way to revert search results so that the packages with the most votes appear at the bottom? Basically [`BottomUp` from `paru`](https://github.com/Morganamilo/paru/blob/2e8c374055b37f76762fd01432d372407c3d8be9/README.md?plain=1#L36). This [workaround](https://github.com/fosskers/aura/pull/159#issuecomment-29560241)... works, but we miss syntax highlighting and it's not very convenient to have to remember to use a pipe every time.

5. `aura -L` [uses `less`](https://fosskers.github.io/aura/log.html?highlight=less#view-the-log-file). Is there a way to use another program? 

6. Likewise, is there a way to use a different pager for [`-k`](https://fosskers.github.io/aura/aur.html?highlight=PKGBUILD#displaying-pkgbuild-changes)? (I haven't tried this flag yet).

7. Can we automatically import PGP keys from `validpgpkeys` field of `PKGBUILD`? couldn't find any info in the manual or the user guide on this.

8. Can we display unread Arch Linux news? same, couldn't find any info about this so I guess it's not possible.

The package set snapshots feature ([`-B`](https://fosskers.github.io/aura/snapshots.html?highlight=snapshot#saving-package-set-snapshots)) is amazing. :rocket:

Thanks for reading!

**Comment by fosskers** [comment]
Thanks for trying out Aura! Just to quickly confirm, are you using Aura 3 or alpha-stage Aura 4? You can find out via `aura -V`. Otherwise, here are answers to your questions:

1. With [Aura 4](https://crates.io/crates/aura-pm), the default config location is `XDG_CONFIG_HOME/aura.toml`. **I have not yet updated the Aura Book to reflect this information**, so apologies for the confusion.
2. That's a great suggestion, I will look into it.
3. This will be more possible in Aura 4, since it caches more information locally.
4. That should be trivial to implement, thanks for the suggestion.
5. Aura 4 uses `bat` by default and `less` if `bat` can't be found. I could also make it possible to specify your own in config.
6. Same as the above.
7. Let me look into this - Aura doesn't have any special handling of keys at the moment.
8. If you'd like that implemented, I can look into it.

**Reply by xfzv** [reply]
Thanks for the details.

> Just to quickly confirm, are you using Aura 3 or alpha-stage Aura 4

I'm using 3.2.9. I'll keep an eye on v4, looks promising! :+1:
